tlslite/tlsconnection.py

```diff
@@ -1542,6 +1577,23 @@ def _serverGetClientHello(self, settings, certChain, verifierDB,
                 else:
                     ffgroup_intersect = [GroupName.ffdhe2048]
 
+        # Check and save clients heartbeat extension mode
+        heartbeatExt = clientHello.getExtension(ExtensionType.heartbeat)
+        if heartbeatExt:
+            if heartbeatExt.mode == HeartbeatMode.peer_allowed_to_send and \
+                    settings.heartbeatResponseCallback:
+                self.heartbeatCanSend = True
+                self.heartbeatResponseCallback = settings.\
+                    heartbeatResponseCallback
+            elif heartbeatExt.mode == HeartbeatMode.\
+                    peer_not_allowed_to_send or not settings.\
+                    heartbeatResponseCallback:
+                self.heartbeatCanSend = False
+            else:
+                for result in self._sendError(
+                        AlertDescription.illegal_parameter,
+                        "Client sent invalid Heartbeat extension"):
```

Comment:
the problem tough is when the client sent "peer_allowed_to_send" but the callback is not set, then the exception is raised...